### PR TITLE
google-cloud-cpp: fix build by overriding abseil-cpp cxxStandard

### LINKED ISSUE #169134: `google-cloud-cpp` build failure on x86_64-linux as of `891d0226`
## Issue description
Build of `google-cloud-cpp` failed on x86_64-linux as of 891d0226de6e281422a8d0d99814925fd7aed8d0. This is currently breaking python3Packages.optax.

```
error: builder for '/nix/store/q3ckz4zh2j25dxxhgzridvfdhvp690dg-google-cloud-cpp-1.38.0.drv' failed with exit code 1;
       last 10 log lines:
       > [852/2269] Linking CXX executable google/cloud/options_benchmark
       > FAILED: google/cloud/options_benchmark
       > : && /nix/store/58pwclg9yr437h0pfgrnbd0jis8fqasd-gcc-wrapper-11.2.0/bin/g++ -O3 -DNDEBUG  google/cloud/CMakeFiles/common_options_benchmark.dir/options_benchmark.cc.o -o google/cloud/options_benchmark  google/cloud/libgoogle_cloud_cpp_common.so.1.38.0  /nix/store/3lw7ca129dwhis7r0ygzkhqf1jqhsvv6-gbenchmark-1.6.1/lib/libbenchmark_main.a  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_bad_optional_access.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_time.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_civil_time.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_strings.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_int128.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_strings_internal.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_base.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_spinlock_wait.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_throw_delegate.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_time_zone.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_bad_variant_access.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_raw_logging_internal.so.2103.0.1  /nix/store/8c7qpymdddn3a23n92sx3nvz7qx6f70j-abseil-cpp-20210324.2/lib/libabsl_log_severity.so.2103.0.1  /nix/store/3lw7ca129dwhis7r0ygzkhqf1jqhsvv6-gbenchmark-1.6.1/lib/libbenchmark.a  -lrt && :
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: google/cloud/libgoogle_cloud_cpp_common.so.1.38.0: undefined reference to `absl::lts_20210324::FormatTime[abi:cxx11](absl::lts_20210324::string_view, absl::lts_20210324::Time, absl::lts_20210324::TimeZone)'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: google/cloud/libgoogle_cloud_cpp_common.so.1.38.0: undefined reference to `absl::lts_20210324::string_view::find(absl::lts_20210324::string_view, unsigned long) const'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: google/cloud/libgoogle_cloud_cpp_common.so.1.38.0: undefined reference to `absl::lts_20210324::ByChar::Find(absl::lts_20210324::string_view, unsigned long) const'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: google/cloud/libgoogle_cloud_cpp_common.so.1.38.0: undefined reference to `absl::lts_20210324::strings_internal::CatPieces[abi:cxx11](std::initializer_list<absl::lts_20210324::string_view>)'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: google/cloud/libgoogle_cloud_cpp_common.so.1.38.0: undefined reference to `absl::lts_20210324::ParseTime(absl::lts_20210324::string_view, absl::lts_20210324::string_view, absl::lts_20210324::Time*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
       > collect2: error: ld returned 1 exit status
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/q3ckz4zh2j25dxxhgzridvfdhvp690dg-google-cloud-cpp-1.38.0.drv'.
error: 1 dependencies of derivation '/nix/store/qmnpxrpmh2a1bvd4sv9h3llslrz47wc6-arrow-cpp-7.0.0.drv' failed to build
building '/nix/store/h194vv9qddklp89ncdfisjcvpgj8jflq-mirrors-list.drv'...
building '/nix/store/ch6jgjnqviqvhysdx0dl924v75733qlj-python3.9-dm-tree-unstable-2021-12-20.drv'...
error: 1 dependencies of derivation '/nix/store/g5w9mpj5r4swalfcy9kdj3jqsfscjpm7-python3.9-pyarrow-7.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/p70222yd3mcjiqr7wys4csirpwhqky28-python3.9-apache-beam-2.37.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/a8m3088chlm9bs0bnzzmn8laf2al4n9a-python3.9-tensorflow-datasets-4.5.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xjgaf3h89jvpanlry1a57m2s2wl158iv-python3.9-optax-0.1.1.drv' failed to build
```

[full build log](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2181534947)

python3Packages.optax maintainers cc: @ndl
google-cloud-cpp maintainers cc: @cpcloud
Other cc: n/a

### Steps to reproduce
1. Checkout nixpkgs at commit 891d0226de6e281422a8d0d99814925fd7aed8d0
2. Run `nix-build -A google-cloud-cpp`

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.13.0-1021-azure, Ubuntu, 20.04.4 LTS (Focal Fossa), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.7.0`
 - nixpkgs: `/nix/store/r902mrj2iwsjhy8gs6y0rbr9nada0z4z-source`
```

## Misc.
This issue was automatically generated by [nixpkgs-upkeep](https://github.com/samuela/nixpkgs-upkeep).
- [CI workflow](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2181534947) that created this issue.
- Internal tag: 6261127071343039327

### PR DESCRIPTION
**Pull request by fabianhjr**
###### Description of changes

Fixes #169134

Didn't test this package nor dependencies. Would appreciate a review @samuela.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by samuela** [comment]
Result of `nixpkgs-review pr 169220` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.spacy-transformers</li>
  </ul>
</details>
<details>
  <summary>29 packages failed to build:</summary>
  <ul>
    <li>arrow-cpp</li>
    <li>parquet-tools</li>
    <li>python310Packages.datafusion</li>
    <li>python310Packages.datasets</li>
    <li>python310Packages.google-cloud-bigquery</li>
    <li>python310Packages.ibis-framework</li>
    <li>python310Packages.pyarrow</li>
    <li>python310Packages.snowflake-connector-python</li>
    <li>python310Packages.snowflake-sqlalchemy</li>
    <li>python310Packages.tokenizers</li>
    <li>python310Packages.transformers</li>
    <li>python39Packages.apache-beam</li>
    <li>python39Packages.datafusion</li>
    <li>python39Packages.datasets</li>
    <li>python39Packages.elegy</li>
    <li>python39Packages.flax</li>
    <li>python39Packages.google-cloud-bigquery</li>
    <li>python39Packages.ibis-framework</li>
    <li>python39Packages.intake</li>
    <li>python39Packages.intake-parquet</li>
    <li>python39Packages.optax</li>
    <li>python39Packages.pyarrow</li>
    <li>python39Packages.snowflake-connector-python</li>
    <li>python39Packages.snowflake-sqlalchemy</li>
    <li>python39Packages.tensorflow-datasets</li>
    <li>python39Packages.tokenizers</li>
    <li>python39Packages.transformers</li>
    <li>python39Packages.treex</li>
    <li>streamlit</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>google-cloud-cpp</li>
  </ul>
</details>

**Comment by samuela** [comment]
Failures appear to be unrelated:
```
error: builder for '/nix/store/j15zhgi9kvbl3zxamvvwpwbgvgg69xy7-python3.9-pytest-mpl-0.14.0.drv' failed with exit code 1;
       last 10 log lines:
       > removing build/bdist.linux-x86_64/wheel
       > Finished executing setuptoolsBuildPhase
       > installing
       > Executing pipInstallPhase
       > /build/pytest-mpl-0.14.0/dist /build/pytest-mpl-0.14.0
       > Processing ./pytest_mpl-0.0.0-py3-none-any.whl
       > Requirement already satisfied: matplotlib in /nix/store/n9xgh1n641a7fcmyp7ll1ard68yy7549-python3.9-matplotlib-3.5.1/lib/python3.9/site-packages (from pytest-mpl==0.0.0) (3.5.1)
       > ERROR: Could not find a version that satisfies the requirement Jinja2 (from pytest-mpl) (from versions: none)
       > ERROR: No matching distribution found for Jinja2
       >
       For full logs, run 'nix log /nix/store/j15zhgi9kvbl3zxamvvwpwbgvgg69xy7-python3.9-pytest-mpl-0.14.0.drv'.
error: 1 dependencies of derivation '/nix/store/ah26agp4qd9z664ki7r3i34b96vy0d9y-python3.9-colorcet-3.0.0.drv' failed to build
error: builder for '/nix/store/2y09bv03180v4284cardhis8wp14pvd6-python3.9-pydeck-0.7.1.drv' failed with exit code 1;
       last 10 log lines:
       >     dist.parse_config_files()
       >   File "/nix/store/xi8ha1k3mjpmnmmi5fp7fmq41f2wwghm-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/dist.py", line 854, in parse_config_files
       >     pyprojecttoml.apply_configuration(self, filename, ignore_option_errors)
       >   File "/nix/store/xi8ha1k3mjpmnmmi5fp7fmq41f2wwghm-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/config/pyprojecttoml.py", line 55, in apply_configuration
       >     config = read_configuration(filepath, True, ignore_option_errors, dist)
       >   File "/nix/store/xi8ha1k3mjpmnmmi5fp7fmq41f2wwghm-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/config/pyprojecttoml.py", line 121, in read_configuration
       >     validate(subset, filepath)
       >   File "/nix/store/xi8ha1k3mjpmnmmi5fp7fmq41f2wwghm-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/config/pyprojecttoml.py", line 44, in validate
       >     raise error from None
       > ValueError: invalid pyproject.toml config: `project`
       For full logs, run 'nix log /nix/store/2y09bv03180v4284cardhis8wp14pvd6-python3.9-pydeck-0.7.1.drv'.
error: builder for '/nix/store/ccns6f7rylia5s9pxk7prn33p90f2pym-python3.9-panel-0.12.7.drv' failed with exit code 1;
       last 10 log lines:
       > installing
       > Executing pipInstallPhase
       > /build/dist /build
       > Processing ./panel-0.12.7-py2.py3-none-any.whl
       > Requirement already satisfied: bleach in /nix/store/igb6rqgg6jppf12xckrl0vk0ibgmq71d-python3.9-bleach-4.1.0/lib/python3.9/site-packages (from panel==0.12.7) (4.1.0)
       > Requirement already satisfied: pyviz-comms>=0.7.4 in /nix/store/nai08697bikd1l4c81dvr2k2fi6vm99w-python3.9-pyviz_comms-2.2.0/lib/python3.9/site-packages (from panel==0.12.7) (2.2.0)
       > Requirement already satisfied: param>=1.10.0 in /nix/store/zkyb0wqrgv6jpbmjjx7gpr77ghnds3s8-python3.9-param-1.12.1/lib/python3.9/site-packages (from panel==0.12.7) (1.12.1)
       > ERROR: Could not find a version that satisfies the requirement setuptools<61,>=42 (from panel) (from versions: none)
       > ERROR: No matching distribution found for setuptools<61,>=42
       >
       For full logs, run 'nix log /nix/store/ccns6f7rylia5s9pxk7prn33p90f2pym-python3.9-panel-0.12.7.drv'.
error: 2 dependencies of derivation '/nix/store/arbs5rw098gcal1m7ffx468gxq5fna87-python3.9-holoviews-1.14.8.drv' failed to build
error: 2 dependencies of derivation '/nix/store/01v4l9kvvlmrk0a2byfqah9sgvc3baah-python3.9-hvplot-0.7.3.drv' failed to build
error: builder for '/nix/store/ch6jgjnqviqvhysdx0dl924v75733qlj-python3.9-dm-tree-unstable-2021-12-20.drv' failed with exit code 1;
       last 10 log lines:
       >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
       >   File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
       >   File "<frozen importlib._bootstrap_external>", line 850, in exec_module
       >   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
       >   File "/nix/store/knkssclwjkw788p10zald4h1ln6pjcpc-python3.9-dm-tree-unstable-2021-12-20/lib/python3.9/site-packages/tree/__init__.py", line 23, in <module>
       >     from .sequence import _is_attrs
       >   File "/nix/store/knkssclwjkw788p10zald4h1ln6pjcpc-python3.9-dm-tree-unstable-2021-12-20/lib/python3.9/site-packages/tree/sequence.py", line 19, in <module>
       >     from tree import _tree
       > ImportError: /nix/store/knkssclwjkw788p10zald4h1ln6pjcpc-python3.9-dm-tree-unstable-2021-12-20/lib/python3.9/site-packages/tree/_tree.cpython-39-x86_64-linux-gnu.so: undefined symbol: _ZN4absl12lts_2021032416strings_internal9CatPiecesB5cxx11ESt16initializer_listI
NS0_11string_viewEE
       For full logs, run 'nix log /nix/store/ch6jgjnqviqvhysdx0dl924v75733qlj-python3.9-dm-tree-unstable-2021-12-20.drv'.
error: 1 dependencies of derivation '/nix/store/rl14f57qjrzjfp3hz6kn18lc2v9iakj9-python3.9-chex-0.1.2.drv' failed to build
error: 2 dependencies of derivation '/nix/store/w2gqlk0r30rcglffghl3nsx5fpkg0npb-python3.9-dm-haiku-0.0.6.drv' failed to build
error: builder for '/nix/store/f1k2593w12pp4i3gzvhd40ykl31vds20-aws-sdk-cpp-1.9.150.drv' failed with exit code 2;
       last 10 log lines:
       > [ 96%] Built target aws-cpp-sdk-rds
       > [ 96%] Linking CXX shared library libaws-cpp-sdk-ssm.so
       > [ 96%] Built target aws-cpp-sdk-ssm
       > [ 96%] Linking CXX shared library libaws-cpp-sdk-iot.so
       > [ 96%] Built target aws-cpp-sdk-iot
       > [ 96%] Linking CXX shared library libaws-cpp-sdk-sagemaker.so
       > [ 96%] Built target aws-cpp-sdk-sagemaker
       > [ 96%] Linking CXX shared library libaws-cpp-sdk-ec2.so
       > [ 96%] Built target aws-cpp-sdk-ec2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/f1k2593w12pp4i3gzvhd40ykl31vds20-aws-sdk-cpp-1.9.150.drv'.
error: 1 dependencies of derivation '/nix/store/21qf4jikrp858y5ck6c7pkbpxzpfinm5-arrow-cpp-7.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/sy4x1kqqxnps7zv5z6zjzdbj79msr2h6-arrow-cpp-7.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4k4rw39l88rsl3nyv42a2fk4z3bszvss-python3.10-pyarrow-7.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4p2a26wz7a2s5pcr6iig788zqqg1cd17-python3.9-pyarrow-7.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2ws099c6q0icfr46i7n0sycqzffxbgbd-parquet-tools-0.2.9.drv' failed to build
error: 1 dependencies of derivation '/nix/store/58m4bvqk3ha7d757f97zzkfwa0f61dii-python3.10-datafusion-0.4.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9pb00zqln4dszbdbh6l2kcw87dnhqj8m-python3.10-datasets-1.18.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hg892m741l21d1h0gp0c7dzy45lm8d4k-python3.10-google-cloud-bigquery-3.0.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vn9cgcqz0nnsi6mqyzsjxjw1v4gcb1jz-python3.10-ibis-framework-2.1.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/nd9lvjlwvia3az1z48sc5yfv8r2iyp3s-python3.10-snowflake-connector-python-2.7.6.drv' failed to build
error: 1 dependencies of derivation '/nix/store/s335pxsjyagqa1wwgbyzvqiiq33flr1i-python3.9-apache-beam-2.37.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/nxhjd48z36kzkka540mwck6n2dv5sxrv-python3.9-datafusion-0.4.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6ajs3yzkiqpyigf6gz460a2qj3cd7jfh-python3.9-datasets-1.18.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xrsy2gyrkfj9cy8468j4ybinj859i4q8-python3.9-google-cloud-bigquery-3.0.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rsy6dmlaakcmd09mczvwqdpa7zl0z0hh-python3.9-ibis-framework-2.1.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/dyzjhvhq4ddcms0bkcw2d2wymr51hkws-python3.9-intake-parquet-0.2.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/n89rcp6b8niqd5z2124ccmknm8iblvkq-python3.9-snowflake-connector-python-2.7.6.drv' failed to build
error: 2 dependencies of derivation '/nix/store/0qhr2kkihq8qi40rw5508pinwdxja13l-streamlit-1.2.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/f0ss4qa7zfpxpqrb83ynfzy98j0lkx2d-python3.10-snowflake-sqlalchemy-1.3.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/gx2jir19w20ha3712dbkyq2g2qing51z-python3.10-tokenizers-unstable-2021-08-13.drv' failed to build
error: 5 dependencies of derivation '/nix/store/12v44dah18xgrqxcs77l2aj7vlzlly4s-python3.9-intake-0.6.4.drv' failed to build
error: 1 dependencies of derivation '/nix/store/7w5mw3nk5l40r6y33vz2za0dcbrm21rd-python3.9-snowflake-sqlalchemy-1.3.3.drv' failed to build
error: 2 dependencies of derivation '/nix/store/6x5c0yc5r1p372nfcyal3sh090r4j7kl-python3.9-tensorflow-datasets-4.5.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/k00y1jj8nw36i3bbfpgh7pvjs8cn6n17-python3.9-tokenizers-unstable-2021-08-13.drv' failed to build
error: 1 dependencies of derivation '/nix/store/zk2634d1sg314v8012z093icyxcprnjf-python3.10-transformers-4.16.2.drv' failed to build
error: 3 dependencies of derivation '/nix/store/wcx65hgbdxz36s43faaqcrb8m5h3g88r-python3.9-optax-0.1.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/q02z0vh0n63axavkqjjch2njmip20ym4-python3.9-transformers-4.16.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/kaq35ykjjgx7v5dkh4wq2k1cx9s7lcx1-python3.9-flax-0.4.1.drv' failed to build
error: 2 dependencies of derivation '/nix/store/92d9bk0ql7g1ldwjaqp7svapia5f68yh-python3.9-treex-0.6.10.drv' failed to build
error: 2 dependencies of derivation '/nix/store/8rij19knmyksb218yanfrm6rw3k0gjfz-python3.9-elegy-0.8.6.drv' failed to build
error: 29 dependencies of derivation '/nix/store/bpfgcanbng8skp3673fmwngary7xhc4i-review-shell.drv' failed to build
```

**Review by samuela — APPROVED** [review]
changes LGTM... lots of things are still broken but fixing google-cloud-cpp is def a step forward

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -6431,7 +6431,11 @@ with pkgs;
 
   google-guest-oslogin = callPackage ../tools/virtualization/google-guest-oslogin { };
 
-  google-cloud-cpp = callPackage ../development/libraries/google-cloud-cpp { };
+  google-cloud-cpp = callPackage ../development/libraries/google-cloud-cpp {
+    abseil-cpp = abseil-cpp.override {
+      cxxStandard = "14";
+    };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:6437]: I think we should add a top-level entry for abseil-ccp-cxx-14.

**Pull request merged**